### PR TITLE
feature/POK-49-implement-keep-session-alive

### PR DESCRIPTION
**Pull request by jimibilibob**
- Add Storage interface and its implementations for iOS and Android.

- Modified the IocService to register new Types.

- Register the dependencies using Unity Container.

### CODE REVIEW

File: PokeAppAndroid/AndroidApplication.cs [hunk 1]
```diff
@@ -16,7 +24,11 @@ public AndroidApplication(IntPtr javaReference, Android.Runtime.JniHandleOwnersh
         public override void OnCreate()
         {
             base.OnCreate();
-			IocContainer.CreateContainer();
+
+            // Storage Utils
+            IocContainer.CreateContainer();
+            IocContainer.RegisterType<IStorage, Storage>();
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply PokeAppAndroid/AndroidApplication.cs:30]: It is not possible, as the Storage class has it's own implementation in both patforms.

File: PokeAppiOS/AppDelegate.cs [hunk 2]
```diff
@@ -17,6 +20,8 @@ public class AppDelegate : UIResponder, IUIApplicationDelegate {
         public bool FinishedLaunching (UIApplication application, NSDictionary launchOptions)
         {
             IocContainer.CreateContainer();
+            IocContainer.RegisterType<IStorage, Storage>();
+            IocContainer.RegisterType<IStorageUtils, StorageUtils>();
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply PokeAppiOS/AppDelegate.cs:24]: It is not possible, as the Storage class has it's own implementation in both patforms.

File: SharedCode/DI/DI.cs [hunk 3]
```diff
@@ -23,6 +25,12 @@ public static void RegisterDIs()
 			container.RegisterType<IPokemonController, PokemonController>();
 			container.RegisterType<IPokemonDetailController, PokemonDetailController>();
 		}
-	}
+
+		public static void RegisterType<I, T>() where T : I
+        {
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply SharedCode/DI/DI.cs]: I don't get it.

[the remark this answers is not in the document]
Comment (jimibilibob) [reply SharedCode/DI/DI.cs]: Everything looks aligned in VS.

Screenshot 2023-04-03 at 19 06 25

**Review by rodrigoschar — APPROVED** [review]

**Review by adCBueno — APPROVED** [review]

File: SharedCode/Services/IocService.cs [hunk 4]
```diff
@@ -25,5 +25,10 @@ public static T GetDependency<T>()
         {
             return container.Resolve<T>();
         }
+
+        public static void RegisterType<I, T>() where T : I
+        {
+            DI.DI.RegisterType<I, T>();
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply SharedCode/Services/IocService.cs]: You're right. It's changed in last commit.

File: SharedCode/Util/StorageUtils.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        private IStorage Storage;
+
+        public StorageUtils(IStorage storage)
+        {
+            this.Storage = storage;
+        }
+
+        public bool GetIsLoggedIn()
+        {
+            return Storage.Read("isLoggedIn", false);
```
Comment (YawarValeriano) [inline SharedCode/Util/StorageUtils.cs]: Maybe define the string as a constant value would be better so it'll be easier to change in the future

**Review by YawarValeriano — CHANGES_REQUESTED** [review]

File: SharedCode/Util/StorageUtils.cs (hunk 5 above)
Comment (jimibilibob) [reply SharedCode/Util/StorageUtils.cs]: Changed.

**Review by YawarValeriano — APPROVED** [review]

**Pull request merged**